jobs/resource_job.go

```diff
@@ -445,6 +445,184 @@ type JobsAPI struct {
 	context context.Context
 }
 
+type JobSettingsResource struct {
+	jobs.JobSettings
+
+	// BEGIN Jobs API 2.0
+	ExistingClusterID      string               `json:"existing_cluster_id,omitempty" tf:"group:cluster_type"`
+	NewCluster             *compute.ClusterSpec `json:"new_cluster,omitempty" tf:"group:cluster_type"`
+	NotebookTask           *NotebookTask        `json:"notebook_task,omitempty" tf:"group:task_type"`
+	SparkJarTask           *SparkJarTask        `json:"spark_jar_task,omitempty" tf:"group:task_type"`
+	SparkPythonTask        *SparkPythonTask     `json:"spark_python_task,omitempty" tf:"group:task_type"`
+	SparkSubmitTask        *SparkSubmitTask     `json:"spark_submit_task,omitempty" tf:"group:task_type"`
+	PipelineTask           *PipelineTask        `json:"pipeline_task,omitempty" tf:"group:task_type"`
+	PythonWheelTask        *PythonWheelTask     `json:"python_wheel_task,omitempty" tf:"group:task_type"`
+	DbtTask                *DbtTask             `json:"dbt_task,omitempty" tf:"group:task_type"`
+	RunJobTask             *RunJobTask          `json:"run_job_task,omitempty" tf:"group:task_type"`
+	Libraries              []compute.Library    `json:"libraries,omitempty" tf:"alias:library"`
+	TimeoutSeconds         int32                `json:"timeout_seconds,omitempty"`
+	MaxRetries             int32                `json:"max_retries,omitempty"`
+	MinRetryIntervalMillis int32                `json:"min_retry_interval_millis,omitempty"`
+	RetryOnTimeout         bool                 `json:"retry_on_timeout,omitempty"`
+	// END Jobs API 2.0
+}
+
+func (JobSettingsResource) Aliases() map[string]map[string]string {
+	aliases := map[string]map[string]string{
+		"jobs.JobSettingsResource": {
+			"tasks":        "task",
+			"parameters":   "parameter",
+			"job_clusters": "job_cluster",
+			"environments": "environment",
+		},
+		"jobs.GitSource": {
+			"git_url":      "url",
+			"git_provider": "provider",
+			"git_branch":   "branch",
+			"git_tag":      "tag",
+			"git_commit":   "commit",
+		},
+		"jobs.Task": {
+			"libraries": "library",
+		},
+	}
+	return aliases
+}
+
+func (JobSettingsResource) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+	// ======= Replicate what was originally in CustomizeSchema call back func in the manually written schema. ======
```

Comment:
This probably doesn't need to be here as a comment in source control, but it could be a comment on the PR :)